### PR TITLE
chore(fixer): improve fixer logic and include more

### PR DESCRIPTION
**Pull request by MrCloudSec**
### Description

• Add 8 fixers
• Add fixer config file among `--config-fixer` flag.
• Improve fixer execution
• Add documentation

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/tutorials/fixer.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+<img src="../img/fixer.png">
+
+???+ note
+    You can see all the available fixes for each provider with the `--list-fixers` flag.
+
+    ```sh
+    prowler <provider> --list-fixer
+    ```
+
+## Writing a Fixer
```
Comment (jfagoagas) [inline docs/tutorials/fixer.md]:
```suggestion
## Writing a Fixer
```

Comment (jfagoagas) [reply docs/tutorials/fixer.md]: Could you please move this to the Developer Guide?

File: docs/tutorials/fixer.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    except Exception as error:
+        logger.error(
+            f"{error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
+        )
+        return False
+    else:
+        return True
+```
+
+## Fixer Config file
```
Comment (jfagoagas) [inline docs/tutorials/fixer.md]:
```suggestion
## Fixer Config file
```

File: docs/tutorials/fixer.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+???+ note
+    You can see all the available fixes for each provider with the `--list-fixers` flag.
+
+    ```sh
+    prowler <provider> --list-fixer
+    ```
+
+## Writing a Fixer
+To write a fixer, you need to create a file called `<check_id>_fixer.py` inside the check folder, with a function called `fixer` that receives either the region or the resource to be fixed as a parameter, and returns a boolean value indicating if the fix was successful or not.
```
Comment (jfagoagas) [inline docs/tutorials/fixer.md:18]: What if the fixer needs the `region` and the `resource_id`?

File: prowler/lib/check/check.py [hunk 4]
```diff
@@ -462,15 +468,18 @@ def run_check(check: Check, output_options) -> list:
         return findings
 
 
-def run_fixer(check_findings: list):
+def run_fixer(check_findings: list) -> int:
```
Comment (jfagoagas) [inline prowler/lib/check/check.py:471]: Do we have tests for this function? If not please add some.

File: prowler/lib/check/check.py [hunk 5]
```diff
@@ -483,18 +492,34 @@ def run_fixer(check_findings: list):
...
+                            # Check if fixer has region as argument to check if it is a region specific fixer
+                            if "region" in fixer.__code__.co_varnames:
+                                print(
+                                    f"\t{orange_color}FIXING{Style.RESET_ALL} {finding.region}... "
+                                )
+                                if fixer(finding.region):
+                                    fixed_findings += 1
+                                    print(f"\t{Fore.GREEN}DONE{Style.RESET_ALL}")
+                                else:
+                                    print(f"\t{Fore.RED}ERROR{Style.RESET_ALL}")
```
Comment (jfagoagas) [inline prowler/lib/check/check.py:512]: Can you move this to a function not to repeat it? Something like `print_fixer_result`.

File: prowler/lib/cli/parser.py [hunk 6]
```diff
@@ -332,6 +333,12 @@ def __init_config_parser__(self):
             default=default_config_file_path,
             help="Set configuration file path",
         )
+        config_parser.add_argument(
+            "--config-fixer",
```
Comment (jfagoagas) [inline prowler/lib/cli/parser.py]:
```suggestion
            "--fixer-config",
```

I think it'd be better to change the order.

Comment (jfagoagas) [reply prowler/lib/cli/parser.py]: You are also storing this in a `fixer_config` attribute in the provider.

File: prowler/providers/aws/services/iam/iam_password_policy_expires_passwords_within_90_days_or_less/iam_password_policy_expires_passwords_within_90_days_or_less_fixer.py [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+                "Action": "iam:UpdateAccountPasswordPolicy",
+                "Resource": "*"
+            }
+        ]
+    }
+    Returns:
+        bool: True if IAM password policy is updated, False otherwise
+    """
+    try:
+        iam_client.client.update_account_password_policy(
```
Comment (jfagoagas) [inline prowler/providers/aws/services/iam/iam_password_policy_expires_passwords_within_90_days_or_less/iam_password_policy_expires_passwords_within_90_days_or_less_fixer.py:23]: I don't get the point of doing the same in all these new fixers, if you just want to fix the `MaxPasswordAge` you have to leave the rest of the values as the AWS account has configured. You have this values on the client so it wouldn't be difficult.

File: prowler/providers/aws/services/iam/iam_password_policy_lowercase/iam_password_policy_lowercase_fixer.py [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+                "Action": "iam:UpdateAccountPasswordPolicy",
+                "Resource": "*"
+            }
+        ]
+    }
+    Returns:
+        bool: True if IAM password policy is updated, False otherwise
+    """
+    try:
+        iam_client.client.update_account_password_policy(
```
Comment (jfagoagas) [inline prowler/providers/aws/services/iam/iam_password_policy_lowercase/iam_password_policy_lowercase_fixer.py:23]: I don't get the point of doing the same in all these new fixers, if you just want to fix the `RequireLowercaseCharacters` you have to leave the rest of the values as the AWS account has configured. You have this values on the client so it wouldn't be difficult.

File: prowler/providers/aws/services/iam/iam_password_policy_minimum_length_14/iam_password_policy_minimum_length_14_fixer.py [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+                "Action": "iam:UpdateAccountPasswordPolicy",
+                "Resource": "*"
+            }
+        ]
+    }
+    Returns:
+        bool: True if IAM password policy is updated, False otherwise
+    """
+    try:
+        iam_client.client.update_account_password_policy(
```
Comment (jfagoagas) [inline prowler/providers/aws/services/iam/iam_password_policy_minimum_length_14/iam_password_policy_minimum_length_14_fixer.py:23]: I don't get the point of doing the same in all these new fixers, if you just want to fix the `MinimumPasswordLength` you have to leave the rest of the values as the AWS account has configured. You have this values on the client so it wouldn't be difficult.

File: prowler/providers/aws/services/iam/iam_password_policy_number/iam_password_policy_number_fixer.py [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+                "Action": "iam:UpdateAccountPasswordPolicy",
+                "Resource": "*"
+            }
+        ]
+    }
+    Returns:
+        bool: True if IAM password policy is updated, False otherwise
+    """
+    try:
+        iam_client.client.update_account_password_policy(
```
Comment (jfagoagas) [inline prowler/providers/aws/services/iam/iam_password_policy_number/iam_password_policy_number_fixer.py:23]: I don't get the point of doing the same in all these new fixers, if you just want to fix the `RequireNumbers` you have to leave the rest of the values as the AWS account has configured. You have this values on the client so it wouldn't be difficult.

File: prowler/providers/aws/services/iam/iam_password_policy_reuse_24/iam_password_policy_reuse_24_fixer.py [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+                "Action": "iam:UpdateAccountPasswordPolicy",
+                "Resource": "*"
+            }
+        ]
+    }
+    Returns:
+        bool: True if IAM password policy is updated, False otherwise
+    """
+    try:
+        iam_client.client.update_account_password_policy(
```
Comment (jfagoagas) [inline prowler/providers/aws/services/iam/iam_password_policy_reuse_24/iam_password_policy_reuse_24_fixer.py:23]: I don't get the point of doing the same in all these new fixers, if you just want to fix the `PasswordReusePrevention` you have to leave the rest of the values as the AWS account has configured. You have this values on the client so it wouldn't be difficult.

File: prowler/providers/aws/services/iam/iam_password_policy_symbol/iam_password_policy_symbol_fixer.py [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+                "Action": "iam:UpdateAccountPasswordPolicy",
+                "Resource": "*"
+            }
+        ]
+    }
+    Returns:
+        bool: True if IAM password policy is updated, False otherwise
+    """
+    try:
+        iam_client.client.update_account_password_policy(
```
Comment (jfagoagas) [inline prowler/providers/aws/services/iam/iam_password_policy_symbol/iam_password_policy_symbol_fixer.py:23]: I don't get the point of doing the same in all these new fixers, if you just want to fix the `RequireSymbols` you have to leave the rest of the values as the AWS account has configured. You have this values on the client so it wouldn't be difficult.

File: prowler/providers/aws/services/iam/iam_password_policy_uppercase/iam_password_policy_uppercase_fixer.py [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+                "Action": "iam:UpdateAccountPasswordPolicy",
+                "Resource": "*"
+            }
+        ]
+    }
+    Returns:
+        bool: True if IAM password policy is updated, False otherwise
+    """
+    try:
+        iam_client.client.update_account_password_policy(
```
Comment (jfagoagas) [inline prowler/providers/aws/services/iam/iam_password_policy_uppercase/iam_password_policy_uppercase_fixer.py:23]: I don't get the point of doing the same in all these new fixers, if you just want to fix the `RequireUppercaseCharacters` you have to leave the rest of the values as the AWS account has configured. You have this values on the client so it wouldn't be difficult.

File: prowler/providers/aws/services/s3/s3_account_level_public_access_blocks/s3_account_level_public_access_blocks_fixer.py [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+                "Action": "s3:PutAccountPublicAccessBlock",
+                "Resource": "*"
+            }
+        ]
+    }
+    Returns:
+        bool: True if S3 Block Public Access is enabled, False otherwise
+    """
+    try:
+        s3control_client.client.put_public_access_block(
```
Comment (jfagoagas) [inline prowler/providers/aws/services/s3/s3_account_level_public_access_blocks/s3_account_level_public_access_blocks_fixer.py:23]: Please add a test for this function.

File: prowler/providers/aws/services/iam/iam_password_policy_uppercase/iam_password_policy_uppercase_fixer.py (hunk 13 above)
Comment (jfagoagas) [reply prowler/providers/aws/services/iam/iam_password_policy_uppercase/iam_password_policy_uppercase_fixer.py:23]: Please add a test for this function.

File: prowler/providers/aws/services/iam/iam_password_policy_symbol/iam_password_policy_symbol_fixer.py (hunk 12 above)
Comment (jfagoagas) [reply prowler/providers/aws/services/iam/iam_password_policy_symbol/iam_password_policy_symbol_fixer.py:23]: Please add a test for this function.

File: prowler/providers/aws/services/iam/iam_password_policy_reuse_24/iam_password_policy_reuse_24_fixer.py (hunk 11 above)
Comment (jfagoagas) [reply prowler/providers/aws/services/iam/iam_password_policy_reuse_24/iam_password_policy_reuse_24_fixer.py:23]: Please add a test for this function.

File: prowler/providers/aws/services/iam/iam_password_policy_number/iam_password_policy_number_fixer.py (hunk 10 above)
Comment (jfagoagas) [reply prowler/providers/aws/services/iam/iam_password_policy_number/iam_password_policy_number_fixer.py:23]: Please add a test for this function.

File: prowler/providers/aws/services/iam/iam_password_policy_minimum_length_14/iam_password_policy_minimum_length_14_fixer.py (hunk 9 above)
Comment (jfagoagas) [reply prowler/providers/aws/services/iam/iam_password_policy_minimum_length_14/iam_password_policy_minimum_length_14_fixer.py:23]: Please add a test for this function.

File: prowler/providers/aws/services/iam/iam_password_policy_lowercase/iam_password_policy_lowercase_fixer.py (hunk 8 above)
Comment (jfagoagas) [reply prowler/providers/aws/services/iam/iam_password_policy_lowercase/iam_password_policy_lowercase_fixer.py:23]: Please add a test for this function.

File: prowler/providers/aws/services/iam/iam_password_policy_expires_passwords_within_90_days_or_less/iam_password_policy_expires_passwords_within_90_days_or_less_fixer.py (hunk 7 above)
Comment (jfagoagas) [reply prowler/providers/aws/services/iam/iam_password_policy_expires_passwords_within_90_days_or_less/iam_password_policy_expires_passwords_within_90_days_or_less_fixer.py:23]: Please add a test for this function.

**Review by jfagoagas — CHANGES_REQUESTED** [review]
Thats a great improvement @sergargar !!! 👏 

Please review my comments and add tests for all the new included fixers, thanks!

File: docs/tutorials/fixer.md (hunk 3 above)
Comment (MrCloudSec) [reply docs/tutorials/fixer.md:18]: It is not contemplated, but good point...

File: prowler/lib/cli/parser.py (hunk 6 above)
Comment (MrCloudSec) [reply prowler/lib/cli/parser.py]: Done

File: prowler/providers/aws/services/iam/iam_password_policy_expires_passwords_within_90_days_or_less/iam_password_policy_expires_passwords_within_90_days_or_less_fixer.py [hunk 15]
```diff
@@ -0,0 +1,43 @@
...
+                "Action": "iam:UpdateAccountPasswordPolicy",
+                "Resource": "*"
+            }
+        ]
+    }
+    Returns:
+        bool: True if IAM password policy is updated, False otherwise
+    """
+    try:
+        print(iam_client.fixer_config)
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/iam/iam_password_policy_expires_passwords_within_90_days_or_less/iam_password_policy_expires_passwords_within_90_days_or_less_fixer.py]:
```suggestion
```

**Review by jfagoagas — DISMISSED** [review]
Awesome! Thanks for taking care of the comments 🚀

File: tests/config/fixtures/config.yaml [hunk 16]
```diff
@@ -1,3 +1,5 @@
+# TODO: UPDATE YAML
```
Comment (jfagoagas) [inline tests/config/fixtures/config.yaml:1]: What is this?

Comment (MrCloudSec) [reply tests/config/fixtures/config.yaml:1]: To update the config yaml with the latest changes, meaning that we have to update the tests.

Comment (jfagoagas) [reply tests/config/fixtures/config.yaml:1]: I didn't saw that it was the fixtures one, thanks!

**Review by jfagoagas — APPROVED** [review]

**Pull request merged**